### DISCUSSION TITLE
Can Yabai disable shadows on macOS 15?

### DISCUSSION THREAD

**Discussion by MoltenMonster** [asmvik/yabai#2498] [Q&A]
I didn't look too closely at the instructions, I installed Yabai with homebrew, has SIP disabled (as shown with `csrutil status`), gave it accessibility permission, and did `yabai -m config window_shadow off` but window shadows are still there.

**Comment by rublev** [comment]
@koekeishiya any ideas? trying to get this working for myself on the latest version.

**Comment by Bellavene** [comment]
It is hard to tell what part of the installation process you forgot to apply. Maybe you did not reinstall the scripting additions? My suggestion is to learn the search function of github, because those issues are described many times pretty well.

Anyway try this. Create `yabai-reinstall` somewhere in your path. Paste the code below inside it. Run `chmod +x yabai-reinstall` and finally run the script.

```
#/bin/zsh

set -e

yabai --stop-service
sudo yabai --uninstall-sa || true
sudo yabai --uninstall-service || true

brew uninstall yabai
brew purge yabai
sleep 0.5
brew install yabai

sleep 1

echo "$(\whoami) ALL=(root) NOPASSWD: sha256:$(shasum -a 256 $(\which yabai) | cut -d " " -f 1) $(\which yabai) --load-sa" | sudo tee /etc/sudoers.d/yabai > /dev/null

sudo yabai --load-sa
yabai --install-service
yabai --start-service
```

Don't forget to allow permissions for yabai in macos settings. You may need to restart yabai two-three times `yabai --restart-service` because of permission requests.

Also, don't forget those lines at the beginning of your `.yabairc`

```
yabai -m signal --add event=dock_did_restart action="sudo yabai --load-sa"
sudo yabai --load-sa
yabai -m config debug_output on
```

After enabling `debug_output` you can watch the logs for any errors yabai was stumbled on here
`/tmp/yabai_User.err.log`
`/tmp/yabai_User.out.log`

After you manage yabai to work properly, comment out `yabai -m config debug_output on` with # sign.

**Reply by Bellavene** [reply]
And don't forget to take carefully steps described [here](https://github.com/koekeishiya/yabai/wiki/Disabling-System-Integrity-Protection) to disable SIP

**Reply by Bellavene** [reply]
Oops, changed the code, you don't need HEAD version, it needs a signed certificate.
So for macos 15 it is a bit simpler. Try one more time.
